### PR TITLE
ref(ddm): Switch timing to distribution and update units

### LINKED ISSUE #60171: Fix units on the existing MRIs
We need to go through the codebase and see where the metrics have obviously wrong/missing units. E.g., if the name of the metric contains bytes, it's pretty clear that the unit should not be the default second.

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR switches all `metrics.timing` (whose unit was inferred to not be `second`) to `metrics.distribution`.

The unit will be needed ONLY for the internal metrics product, for this reason, any other metrics backend that we might use in production will still use `metrics.timing` under the hood, to avoid breaking existing production metrics.

A `distribution` left without a unit, defaults to the string `none`.

If you feel like the units I have inferred are not right, feel free to leave a comment in the PR or directly change the unit. That would be really helpful!

Closes https://github.com/getsentry/sentry/issues/60171

### CODE REVIEW

File: src/sentry/consumers/synchronized.py [hunk 1]
```diff
@@ -162,21 +162,23 @@ def assignment_callback(offsets: Mapping[Partition, int]) -> None:
                 remote_offsets[commit.group][commit.partition] = commit.offset
 
             if commit.orig_message_ts is not None:
-                metrics.timing(
+                metrics.distribution(
```
Comment (mitsuhiko) [inline src/sentry/consumers/synchronized.py:165]: Why is this not timing?

File: src/sentry/consumers/synchronized.py [hunk 2]
```diff
@@ -162,21 +162,23 @@ def assignment_callback(offsets: Mapping[Partition, int]) -> None:
...
                     "commit_log_msg_latency",
                     (now - commit.orig_message_ts) * 1000,
                     tags={
                         "partition": str(commit.partition.index),
                         "group": commit.group,
                     },
+                    unit="millisecond",
                 )
-            metrics.timing(
+            metrics.distribution(
```
Comment (mitsuhiko) [inline src/sentry/consumers/synchronized.py:174]: Same here, should be timing.

File: src/sentry/event_manager.py [hunk 3]
```diff
@@ -708,7 +708,7 @@ def save_error_events(
             job["received_timestamp"] - job["recorded_timestamp"],
             tags=metric_tags,
         )
-        metrics.timing("events.size.data.post_save", job["event"].size, tags=metric_tags)
+        metrics.distribution("events.size.data.post_save", job["event"].size, tags=metric_tags)
```
Comment (mitsuhiko) [inline src/sentry/event_manager.py:711]: I think this has a unit of `byte`.

File: src/sentry/utils/services.py [hunk 4]
```diff
@@ -592,7 +592,7 @@ def callback_timing(
             )
         else:
             secondary_duration_ms = (secondary_timing[1] - secondary_timing[0]) * 1000
-            metrics.timing(
+            metrics.distribution(
```
Comment (mitsuhiko) [inline src/sentry/utils/services.py:595]: should stay timing.

File: src/sentry/utils/services.py [hunk 5]
```diff
@@ -601,17 +601,20 @@ def callback_timing(
                     "status": secondary_status,
                     "primary": "false",
                 },
+                unit="millisecond",
                 **metric_kwargs,  # type: ignore
             )
-            metrics.timing(
+            metrics.distribution(
```
Comment (mitsuhiko) [inline src/sentry/utils/services.py:607]: should stay timing.

File: src/sentry/utils/services.py [hunk 6]
```diff
@@ -601,17 +601,20 @@ def callback_timing(
...
                 f"{metric_name}.timing_delta_ms",
                 secondary_duration_ms - primary_duration_ms,
                 tags=tags,
+                unit="millisecond",
                 **metric_kwargs,  # type: ignore
             )
-            metrics.timing(
+            metrics.distribution(
```
Comment (mitsuhiko) [inline src/sentry/utils/services.py:614]: should stay timing.

File: src/sentry/web/frontend/js_sdk_loader.py [hunk 7]
```diff
@@ -211,6 +211,6 @@ def get(
             response["Surrogate-Key"] = f"project/{key.project_id} sdk/{sdk_version} sdk-loader"
 
         ms = int((time.time() - start_time) * 1000)
-        metrics.timing("js-sdk-loader.duration", ms, instance=instance)
+        metrics.distribution("js-sdk-loader.duration", ms, instance=instance, unit="millisecond")
```
Comment (mitsuhiko) [inline src/sentry/web/frontend/js_sdk_loader.py:214]: should stay timing.

**Review by mitsuhiko — DISMISSED** [review]

File: src/sentry/web/frontend/js_sdk_loader.py (hunk 7 above)
Comment (iambriccardo) [reply src/sentry/web/frontend/js_sdk_loader.py:214]: Timing uses `second` not millisecond.

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**